### PR TITLE
Fix: fix double-trim bug in entropy computation for last CP rank

### PR DESCRIPTION
**Pull request by Beichen-Ma**
In `get_logprob_and_entropy_with_cp`, the logits is trimed twice for last rank, causing missing entropy for one token position and incorrect entropy values when using CP with fsdp backend.

Consolidate into single shifted_logits variable used for both log_probs and entropy.

### CODE REVIEW

**Comment by yueming-yuan** [comment]
(this might not be compatible with new FSDP impl...)

**Comment by PopSoda2002** [comment]
I believe it's not a bug, we can sync for this although the implementation is a little bit hardcode
